src/peft/utils/other.py

```diff
@@ -876,30 +889,48 @@ def _set_trainable(
 
 
 def _set_adapter(model, adapter_name):
-    def check_adapter_name(adapter_name):
+    def check_adapter_name(adapter_name: str | list[str], module: AuxiliaryTrainingWrapper) -> str | None:
+        # Helper function to check if the adapter(s) can be set. Right now, only exactly 0 or 1 active adapters are
+        # allowed.
         if isinstance(adapter_name, str):
             return adapter_name
 
         # adapter_name is a list of str
-        if len(adapter_name) > 1:
-            raise ValueError("Only one adapter can be set at a time for modules_to_save")
-        elif len(adapter_name) == 0:
+        if len(adapter_name) == 0:
             raise ValueError("Please specify at least one adapter to set")
-        adapter_name = adapter_name[0]
-        return adapter_name
+
+        adapter_name_to_set: str | None
+        if isinstance(module, TrainableTokensWrapper):
+            # TODO In theory, multiple active trainable tokens is fine when the indices don't overlap
+            adapter_names_in_module = [n for n in adapter_name if n in module.token_adapter.trainable_tokens_delta]
+        elif isinstance(module, ModulesToSaveWrapper):
+            adapter_names_in_module = [n for n in adapter_name if n in module.modules_to_save]
+        else:
+            raise NotImplementedError(f"Please implement the handling for {type(module)}")
+
+        if len(adapter_names_in_module) > 1:
+            raise ValueError(f"Only one adapter can be set at a time for {module}, got {len(adapter_names_in_module)}")
+
+        if not adapter_names_in_module:
+            adapter_name_to_set = None
+        else:
+            adapter_name_to_set = adapter_names_in_module[0]
+
+        return adapter_name_to_set
```

Comment:
I wonder if we should delegate the check (and potential checking of capabilities like merging given a method - out of scope but forseeable) to the wrapper classes themselves, e.g. `AuxiliaryTrainingWrapper.check_new_adapter()`. WDYT?